ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java

```diff
@@ -75,14 +77,25 @@ public void connect(final String url, @Nullable final ReadableArray protocols, @
         .tag(id)
         .url(url);
 
-    if (options != null && options.hasKey("origin")) {
-      if (ReadableType.String.equals(options.getType("origin"))) {
-        builder.addHeader("Origin", options.getString("origin"));
-      } else {
-        FLog.w(
-          ReactConstants.TAG,
-          "Ignoring: requested origin, value not a string");
+    if (headers != null) {
+      ReadableMapKeySetIterator iterator = headers.keySetIterator();
+
+      if (!headers.hasKey("origin")) {
+        builder.addHeader("origin", setDefaultOrigin(url));
       }
+
+      while (iterator.hasNextKey()) {
+        String key = iterator.nextKey();
+        if (ReadableType.String.equals(headers.getType(key))) {
+          builder.addHeader(key, headers.getString(key));
+        } else {
+          FLog.w(
+            ReactConstants.TAG,
+            "Ignoring: requested origin, value not a string");
```

Comment:
`iterator` loops over all the headers, correct? Why log "requested origin" here even if that specific item is unrelated to the origin header?